doc/data-warehouse-what-and-why.md

```diff
@@ -0,0 +1,258 @@
+<!-- markdown-toc start - Don't edit this section. Run M-x markdown-toc-refresh-toc -->
+**Table of Contents**
+
+- [The need for a Data Warehouse](#the-need-for-a-data-warehouse)
+    - [Long term vision](#long-term-vision)
+    - [Give departments the data they need](#give-departments-the-data-they-need)
+    - [At scale and over time](#at-scale-and-over-time)
+        - [Example 1: Number of Content Items published by an organization](#example-1-number-of-content-items-published-by-an-organization)
+        - [Example 2: Enforcing good practices](#example-2-enforcing-good-practices)
+        - [Example 3: Predictions](#example-3-predictions)
+- [Why a Data Warehouse](#why-a-data-warehouse)
+    - [Overview](#overview)
+    - [Track content changes over time for Content Items](#track-content-changes-over-time-for-content-items)
+    - [A new application for GOV.UK?](#a-new-application-for-govuk)
+
+<!-- markdown-toc end -->
+
+# The need for a Data Warehouse
```

Comment:
Is this the right title for the section? Because the following sentence is spent clarifying it. It seems that this section is more about the overall project goals and aims.

Reply:
I have tweaked titles and structure:

```
- [The need for tracking metrics through time](#the-need-for-tracking-metrics-through-time)
    - [Long term vision](#long-term-vision)
    - [Give departments the data they need](#give-departments-the-data-they-need)
    - [At scale and over time](#at-scale-and-over-time)
        - [Example 1: Number of Content Items published by an organization](#example-1-number-of-content-items-published-by-an-organization)
        - [Example 2: Enforcing good practices](#example-2-enforcing-good-practices)
        - [Example 3: Predictions](#example-3-predictions)
- [Why a Data Warehouse](#why-a-data-warehouse)
    - [State of the art](#state-of-the-art)
    - [Tracking metrics through time in a Data Warehouse](#tracking-metrics-through-time-in-a-data-warehouse)
    - [Track content changes over time for Content Items](#track-content-changes-over-time-for-content-items)
```